tls/x509utils/names.go

```diff
@@ -0,0 +1,117 @@
+package x509utils
+
+import (
+	"crypto/x509"
+	"fmt"
+	"net"
+	"net/netip"
+	"net/url"
+	"strings"
+
+	"darvaza.org/core"
+)
+
+// Names returns a list of exact names and patterns the certificate
+// supports
+func Names(cert *x509.Certificate) ([]string, []string) {
+	names, patterns := splitDNSNames(cert.DNSNames)
+	names = appendIPAddresses(names, cert.IPAddresses)
+
+	if len(names) == 0 && len(patterns) == 0 {
+		if len(cert.URIs) == 0 {
+			// an "old" certificate, no SAN
+			cn := cert.Subject.CommonName
+			if cn != "" {
+				names = append(names, strings.ToLower(cn))
+			}
+		}
+	}
+	return names, patterns
+}
```

Comment:
@karasz ---^ ?

Reply:
Yup, it seems that Common Name is to be avoided. Here is the research I did:
1. [Chrome](https://developer.chrome.com/blog/chrome-58-deprecations/) and [Mozilla](https://bugzilla.mozilla.org/show_bug.cgi?id=1245280) both disregard Common Name for a long time now, both citing that Common Name is deprecated since [RFC2818](https://www.rfc-editor.org/rfc/rfc2818) although the wording used in that RFC is not clearly forbids using CN, more like advises against.
2. [This is a longer explanation of the issue](https://frasertweedale.github.io/blog-redhat/posts/2017-07-11-cn-deprecation.html)

In my opinion we should skip CN

Reply:
@karasz amended accordinging
@coderabbitai review